Dockerfile.auth

```
@@ -0,0 +1,83 @@
+# syntax=docker/dockerfile:1.6
+
+ARG RUST_VERSION=1.81
+ARG APP_NAME=calimero-auth
+
+################################################################################
+# Development Stage - this is the stage we'll use for development
+FROM rust:${RUST_VERSION}-slim-bookworm AS dev
+
+WORKDIR /usr/src/calimero
+
+# Install build dependencies
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    cmake build-essential ca-certificates git \
+    libssl-dev pkg-config clang libclang-dev llvm-dev \
+    libc6-dev libsnappy-dev zlib1g-dev libbz2-dev \
+    libgflags-dev liblz4-dev libzstd-dev curl \
+    && rm -rf /var/lib/apt/lists/*
+
+# We don't COPY any source code here - it will be mounted at runtime
+# Copy config file
+COPY crates/auth/config/config.toml /etc/calimero/auth.toml
+
+EXPOSE 3001
+
+# Default command for development
+CMD ["cargo", "run", "--bin", "calimero-auth", "--", "--config", "/etc/calimero/auth.toml", "--verbose"]
+
+################################################################################
+# Production Stage - this builds the final optimized binary
+FROM dev AS build
+ARG APP_NAME
+
+# For production build, copy the source
+COPY . .
+
+# Build the release binary with forced frontend fetch
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
+    --mount=type=cache,target=/usr/local/cargo/git \
+    --mount=type=cache,target=/usr/src/calimero/target \
+    CALIMERO_AUTH_FRONTEND_FETCH=1 cargo build --release --bin ${APP_NAME} && \
```

surely this isn't needed yeah? how about we leave the user to use `--build-arg CALIMERO_AUTH_FRONTEND_FETCH=1` should they need to

```suggestion
    cargo build --release --bin ${APP_NAME} && \
```